aiosqlite/cursor.py

```diff
@@ -2,28 +2,29 @@
 # Licensed under the MIT license
 
 import sqlite3
-from typing import TYPE_CHECKING, Any, Iterable, Optional, Tuple
+from typing import TYPE_CHECKING, Any, AsyncIterator, Iterable, Optional, Tuple
 
 if TYPE_CHECKING:
     from .core import Connection
 
 
 class Cursor:
     def __init__(self, conn: "Connection", cursor: sqlite3.Cursor) -> None:
+        self.iter_chunk_size = conn._iter_chunk_size
         self._conn = conn
         self._cursor = cursor
 
-    def __aiter__(self) -> "Cursor":
+    def __aiter__(self) -> AsyncIterator[sqlite3.Row]:
         """The cursor proxy is also an async iterator."""
-        return self
-
-    async def __anext__(self) -> sqlite3.Row:
-        """Use `cursor.fetchone()` to provide an async iterable."""
-        row = await self.fetchone()
-        if row is None:
-            raise StopAsyncIteration
-
-        return row
+        return self.__fetch_chunked()
+
+    async def __fetch_chunked(self):
```

Comment:
I would still prefer to drop the second underscore, and just have it named `_fetch_chunked`.